pkg/pillar/pubsub/publish.go

```diff
@@ -228,8 +228,12 @@ func (pub *Publication) sendUpdate(sock net.Conn, key string,
 	// base64-encode to avoid having spaces in the key and val
 	sendKey := base64.StdEncoding.EncodeToString([]byte(key))
 	sendVal := base64.StdEncoding.EncodeToString(b)
-	_, err = sock.Write([]byte(fmt.Sprintf("update %s %s %s",
-		pub.topic, sendKey, sendVal)))
+	buf := fmt.Sprintf("update %s %s %s", pub.topic, sendKey, sendVal)
+	if len(buf) >= maxsize {
+		log.Fatalf("Too large message (%d bytes) sent to %s topic %s key %s",
+			len(buf), pub.nameString(), pub.topic, key)
+	}
+	_, err = sock.Write([]byte(buf))
```

Comment:
This part seems to be repeated in 4 different places. Wondering if we are better off refactoring it:

func sockWrite(buf, pub, topic) error {
if len(buf) >= maxsize {
		log.Fatalf("Too large message (%d bytes) sent to %s topic %s key %s",
			len(buf), pub.nameString(), pub.topic, key)
	}
     _, err = sock.Write([]byte(buf))
     return err
}

Reply:
Not identical since we don't have a topic in all cases. And with the duplicated code the log.Fatal will include the function in which it failed; with a separate function it would need to explicitly log which one failed i.e., some additional string from the caller.